### PR TITLE
[Docs] Refine the vignettes

### LINKED ISSUE #185: [Docs] Refine the vignettes
Currently, we have a getting started vignette and three vignettes for `teal.transform`.
Generally, these vignettes are more focussed on the module developers so they do not touch the functions like `choices_selected()` which are used by the teal app developers.
Proposal is to improve the getting started vignette to go in detail about the problem and solution provided by teal.transform and the vignettes will dive deep into these solutions.
Since `teal.transform` is about to be refactored, the code for the examples from these vignettes will still remain the same. The change will be in the layout of the vignettes. With an addition of a new vignette focussed for app developers.

### PR DESCRIPTION
**Pull request by vedhav**
Closes #185 

Changes:
1. Updated the diagram so it's more clear on what teal.transform does.
2. ~Added a small vignette explaining how to create `data_extract_spec`~
3. Update the existing vignette's example code without `teal.widgets` dependency as it's not needed.
4. Added a small note at the top of every vignette warning that the functions might change.
5. Refined the vignettes 
6. Add the screenshot of the app outputs in the examples.

### CODE REVIEW

File: vignettes/app-developer.Rmd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+_Note that `teal.transform` is about to undergo a refactor so the functions in this vignette are subject to change._
+
+The teal app developer has the capability to provide the level of data transformation options in the teal app.
+This vignette provides an overview of how this can be achieved.
+
+### `data_extract_spec` object
+
+All we need to do is provide the required `data_extract_spec` object where it is required by the module.
+This object is used by the module to create the UI which the app user can control and server logic that helps in translating the UI into `R` expressions to get the data requested by the app user.
```
Comment (chlebowa) [inline vignettes/app-developer.Rmd:28]: Are we sure we "R" should be backticked? I don't believe it is common to do to names of languages.

File: vignettes/app-developer.Rmd
Comment (chlebowa) [inline vignettes/app-developer.Rmd:1]: The article needs a proper preface, it jumps into the deep end immediately.

The graph is marginally helpful but I don't quite like its irregular appearance. Can it be shaped so that arguments unfold vertically rather than burst outwards in all directions?

File: vignettes/data-extract-merge.Rmd
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:1]: It seems to me the user has no influence on merging and transforming data whatsoever. The app developer specifies all operations, the can merely subset data that undergoes the transformations.

File: vignettes/data-extract-merge.Rmd [hunk 2]
```diff
@@ -63,34 +72,49 @@ adtte_extract <- data_extract_spec(
...
 }
+```
+
+#### Step 3/5 - Creating the Server Logic
 
+Here, we define the `merge_srv` function, which will be used to create the server logic for the `shiny` app.
+
+This function gets the `datasets` as a list of reactive `data.frame`, the list of `data_extract` objects, and the `join_keys` object as input.
+We make use of the `merge_expression_srv` function to get a reactive list containing merge expression and information needed to perform the transformation - see more in `merge_expression_srv` documentation.
+We render this expression and evaluate it to get the final `ANL` dataset which is also rendered as a table in our UI.
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:115]: What do you mean by rendering an expression?

File: vignettes/data-extract.Rmd [hunk 3]
```diff
@@ -17,13 +17,15 @@ knitr::opts_chunk$set(
 )
 ```
 
+_Note that `teal.transform` is about to undergo a refactor so the functions in this vignette are subject to change._
+
 There are times when an app developer wants to offer users more flexibility in analyzing data within their custom module. 
-In such cases, relinquishing control of the application to users requires developers to provide a degree of freedom. 
+In these situations, the developers need to let the users control the application by giving the users some degree of freedom.
```
Comment (chlebowa) [inline vignettes/data-extract.Rmd:23]: These two sentences are essentially duplicates.

File: vignettes/data-extract.Rmd [hunk 4]
```diff
@@ -17,13 +17,15 @@ knitr::opts_chunk$set(
...
 With `teal`, app developers can open up their applications to users, allowing them to decide exactly which app data to 
 analyze within the module.
 
-Many `teal` modules leverage `data_extract_spec` objects and modules to handle user input.
-Examples can be found in `teal.modules.general` and `teal.modules.clinical`.
+A `teal` module can leverage by using the `data_extract_spec` objects and to handle and process the user input.
+Examples can be found in the `teal.modules.clinical`.
```
Comment (chlebowa) [inline vignettes/data-extract.Rmd:28]:
```suggestion
Examples can be found in the `teal.modules.clinical` package.
```

File: vignettes/data-extract.Rmd [hunk 5]
```diff
@@ -17,13 +17,15 @@ knitr::opts_chunk$set(
...
+
 There are times when an app developer wants to offer users more flexibility in analyzing data within their custom module. 
-In such cases, relinquishing control of the application to users requires developers to provide a degree of freedom. 
+In these situations, the developers need to let the users control the application by giving the users some degree of freedom.
 With `teal`, app developers can open up their applications to users, allowing them to decide exactly which app data to 
 analyze within the module.
 
-Many `teal` modules leverage `data_extract_spec` objects and modules to handle user input.
-Examples can be found in `teal.modules.general` and `teal.modules.clinical`.
+A `teal` module can leverage by using the `data_extract_spec` objects and to handle and process the user input.
```
Comment (chlebowa) [inline vignettes/data-extract.Rmd:27]: leverage what?

**Review by chlebowa — CHANGES_REQUESTED** [review]

File: vignettes/app-developer.Rmd (hunk 1 above)
Comment (vedhav) [reply vignettes/app-developer.Rmd:28]: We used it in [some places](https://insightsengineering.github.io/teal/latest-tag/articles/teal-options.html) earlier. I could go either way. Slightly leaning towards having backtick because it becomes easier to see it.

File: vignettes/data-extract-merge.Rmd (hunk 2 above)
Comment (vedhav) [reply vignettes/data-extract-merge.Rmd:115]:
```suggestion
We evaluate this expression to get the final `ANL` dataset which is also rendered as a table in our UI.
```

How about now?

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd:115]: No wait. I remember what I meant. Let me rephrase it.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd:115]:
```suggestion
We print this expression in the UI  and also evaluate it to get the final `ANL` dataset which is also displayed as a table in the UI.
```

Is this clear now? I wanted to convey that this object is now used to create two outputs in the UI, one is the print output of the expression and the other is the table. It would be easier to see this once we add a screenshot.

File: vignettes/data-extract.Rmd (hunk 3 above)
Comment (vedhav) [reply vignettes/data-extract.Rmd:23]: Agreed, I'll remove this line completely.

File: vignettes/data-extract.Rmd (hunk 5 above)
Comment (vedhav) [reply vignettes/data-extract.Rmd:27]:
```suggestion
A `teal` module can leverage using the `data_extract_spec` objects to handle and process the user input.
```

File: vignettes/app-developer.Rmd
Comment (vedhav) [reply vignettes/app-developer.Rmd:1]: On second thoughts I'm rethinking the diagram now. It would be better to give code snippets of different combinations of valid `des` creations.

File: vignettes/data-extract-merge.Rmd
Comment (vedhav) [reply vignettes/data-extract-merge.Rmd:1]: Yeah, this is true. From what I see, the merge operations are just a function of using the `join_keys` to perform the joins and the app user has nothing to do with this.

File: vignettes/app-developer.Rmd [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (averissimo) [inline vignettes/app-developer.Rmd:1]: Consider just merging the information on this vignette with the existing section on `data_extract_spec` in the "data-extract.Rmd"

It seems like this should be extended somewhat more to be a standalone vignette

Maybe place a section on the other that focus towards app developers.

File: vignettes/app-developer.Rmd [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+```{r setup, include = FALSE, echo=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+_Note that `teal.transform` is about to undergo a refactor so the functions in this vignette are subject to change._
```
[the remark this answers is not in the document]
Comment (averissimo) [reply vignettes/app-developer.Rmd:20]: Agree, this might discourage adoption and we're not 100% sure about when the refactor is going to happen.. next month, next semester, next year?

File: vignettes/data-extract-merge.Rmd [hunk 8]
```diff
@@ -17,20 +17,29 @@ knitr::opts_chunk$set(
...
+
+<img src="images/basic_concept.svg" alt="Basic Concept of teal.transform" style="width: 100%;" />
 
-```{r echo=FALSE, out.width='100%'}
-knitr::include_graphics("./images/data_extract_spec/basic_concept.png")
-```
+
+Let's see how to achieve this dynamic `select`, `filter`, and `merge` operations in a shiny app using `teal.transform`.
+
+#### Step 1/5 - Creating the data extract
```
Comment (averissimo) [inline vignettes/data-extract-merge.Rmd:37]: I like the flow of this document :+1:

File: vignettes/data-extract.Rmd [hunk 9]
```diff
@@ -17,13 +17,14 @@ knitr::opts_chunk$set(
...
+
 There are times when an app developer wants to offer users more flexibility in analyzing data within their custom module. 
-In such cases, relinquishing control of the application to users requires developers to provide a degree of freedom. 
 With `teal`, app developers can open up their applications to users, allowing them to decide exactly which app data to 
 analyze within the module.
 
-Many `teal` modules leverage `data_extract_spec` objects and modules to handle user input.
-Examples can be found in `teal.modules.general` and `teal.modules.clinical`.
+A `teal` module can leverage using the `data_extract_spec` objects to handle and process the user input.
+Examples can be found in the `teal.modules.clinical` package.
```
Comment (averissimo) [inline vignettes/data-extract.Rmd:27]: Add links to tmc documentation page?

File: vignettes/app-developer.Rmd [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+1. Select the required columns from the data.
+2. Filter the data  based on the user's selection.
+3. Reshape the data into long or wide format.
+
+To construct a `data_extract_spec` object use the `data_extract_spec` function and these helper functions: `select_spec`, `filter_spec`, `variable_choices`, and `value_choices`.
+
+You can refer to this diagram to get a quick view of how to use these functions. For more detailed explaination please refer to the function documentation.
+
+<img src="images/data_extract_spec.svg" alt="How to use data_extract_spec()" style="width: 100%;" />
```
Comment (averissimo) [inline vignettes/app-developer.Rmd:42]: This is a very complicated diagram, I'm not sure we can get a quick view of how we use these functions :face_with_spiral_eyes: 

Maybe add a bullet point description of the helper functions to reference them

**Review by averissimo — COMMENTED** [review]
I like the changes, but after seeing the PR I get the feeling the window to get this reviewed and merged is too narrow.

As the vignettes would benefit from more details that would delay the release to CRAN.

On the other hand, adding that detail to deprecated it soon™ may not warrant doing it.

File: vignettes/app-developer.Rmd (hunk 10 above)
Comment (vedhav) [reply vignettes/app-developer.Rmd:42]: > On second thoughts I'm rethinking the diagram now. It would be better to give code snippets of different combinations of valid des creations.

File: vignettes/app-developer.Rmd (hunk 7 above)
[the remark this answers is not in the document]
Comment (vedhav) [reply vignettes/app-developer.Rmd]: Removed them.

File: vignettes/data-extract.Rmd (hunk 9 above)
Comment (vedhav) [reply vignettes/data-extract.Rmd:25]: Sure thing!

File: vignettes/app-developer.Rmd (hunk 1 above)
Comment (vedhav) [reply vignettes/app-developer.Rmd]: Feel free to reopen if you feel strongly about this.

File: vignettes/data-extract-merge.Rmd (hunk 2 above)
Comment (vedhav) [reply vignettes/data-extract-merge.Rmd:113]: This is just explaining what we're doing in the example app created in the vignette. Not the logic behind what is going on. I've added the screenshot of the app now. Does it make things clear now?

File: vignettes/data-extract.Rmd (hunk 5 above)
Comment (vedhav) [reply vignettes/data-extract.Rmd]: does this make sense?

File: vignettes/app-developer.Rmd (hunk 6 above)
Comment (vedhav) [reply vignettes/app-developer.Rmd]: I think we can completely remove this. I added a vignette to talk about `data_extract_spec` thinking that this is the only app developer-facing function. But I feel that this is not the case as tmc does not seem to work with this. So, talking about this seems useless.

**Comment by averissimo** [comment]
> What do you both think about removing the app-developer completely?

Agree! :+1:

**Comment by chlebowa** [comment]
> What do you both think about removing the `app-developer` completely?

Agreed :+1:

File: vignettes/data-extract-merge.Rmd [hunk 11]
```diff
@@ -17,20 +17,27 @@ knitr::opts_chunk$set(
 )
 ```
 
-`teal.transform` provides `merge_expression_srv`, which converts `data_extract_srv` into `R` expressions to transform data 
-for analytical purposes. 
-For example, you may wish to select `AGE` from `ADSL` and select `AVAL` from `ADTTE`, filtered for rows where `PARAMCD` is `OS`, and then merge the results using the primary keys to create an analysis dataset `ANL`.
-This diagram illustrates the concept:
+`teal.transform` provides functions that help in transforming the data based on the shiny app user's input from the UI.
+The UI inputs are translated into `R` expressions to transform the data for analytical purposes.
+This gives the app user flexibility to transform the data as per their requirements.
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:22]:
```suggestion
`teal.transform` allows the app user to oversee transforming a relational set of data objects into the final dataset for analysis.
User actions create a R expression that subsets and merges the input data.
```

File: vignettes/data-extract-merge.Rmd
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:1]: Please don't use contractions (like "don't").
Remember to always backtick package names (here: `shiny`).
Maintain the same case across all section titles.

File: vignettes/data-extract-merge.Rmd [hunk 12]
```diff
@@ -17,20 +17,27 @@ knitr::opts_chunk$set(
...
 
-```{r echo=FALSE, out.width='100%'}
-knitr::include_graphics("./images/data_extract_spec/basic_concept.png")
-```
+For example, let's try to create an analysis dataset `ANL` by:
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:24]: `In the following example we will[...]`

Add somewhere: "Note that primary key columns are maintained when selecting columns from datasets."

File: vignettes/data-extract-merge.Rmd [hunk 13]
```diff
@@ -17,20 +17,27 @@ knitr::opts_chunk$set(
...
+1. Selecting the column `AGE` from `ADSL`
+2. Selecting the column `AVAL` and filtering the rows where `PARAMCD` is `OS` from `ADTTE`
+3. Merging the results from the above datasets using the primary keys.
+
+<img src="images/basic_concept.svg" alt="Basic Concept of teal.transform" style="width: 100%;" />
+
+
+Let's see how to achieve this dynamic `select`, `filter`, and `merge` operations in a shiny app using `teal.transform`.
+
+#### Step 1/5 - Creating the data extract
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:35]: `choices = c("AGE", "BMRKR1"),` limiting choices is not connected to the essence of the example (user selecting a column). Can this be set to all columns? As a non-user: is there an easy way to say "all columns allowed"?

Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:35]:
```suggestion
#### Step 1/5 - Creating data extract specifications
```

What is created is extract _specs_, not data extracts (those will be created in the srv functions and merged). Consider changing variable names accordingly.

File: vignettes/data-extract-merge.Rmd (hunk 2 above)
Comment (chlebowa) [reply vignettes/data-extract-merge.Rmd:113]: Oooh, I see. That's better.

File: vignettes/data-extract-merge.Rmd [hunk 14]
```diff
@@ -63,34 +70,49 @@ adtte_extract <- data_extract_spec(
...
     )
   )
 }
+```
+
+#### Step 3/5 - Creating the Server Logic
+
+Here, we define the `merge_srv` function, which will be used to create the server logic for the `shiny` app.
 
+This function gets the `datasets` as a list of reactive `data.frame`, the list of `data_extract` objects, and the `join_keys` object as input.
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:111]:
```suggestion
This function takes as arguments the datasets (as a list of reactive `data.frame`), the data extract specifications created above (the `data_extract` list), and the `join_keys` object.
```

`join_keys` has not been mentioned before. Add some explanation somewhere or at least change `the` to `a` and add a link to `teal.data` here.

File: vignettes/data-extract-merge.Rmd [hunk 15]
```diff
@@ -111,11 +133,7 @@ merge_srv <- function(id, datasets, data_extracts, join_keys) {
...
 ```
 
-Output from `data_extract_srv` (`reactive`) should be passed to `merge_expression_srv` together with `datasets`
-(list of reactive `data.frame` objects) and `join_keys` object.
-`merge_expression_srv` returns a reactive list containing merge expression and information needed to perform the transformation - see more in `merge_expression_srv` documentation.
-
-#### Example data
+#### Step 4/5 - Preparing the Data
 
 The `data_extract_srv` module depends on a list of reactive or non-reactive `data.frame` objects.
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:138]:
```suggestion
The `data_extract_srv` module takes as argument a list of reactive or non-reactive `data.frame`s. 
```

Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:138]: `data_extract_srv` is not called in this section so I'm not sure the flow is correct.

Maybe preparing data and joining keys should be step 1? It would be a good place to mention `join_keys`.

File: vignettes/data-extract-merge.Rmd [hunk 16]
```diff
@@ -141,9 +159,9 @@ join_keys <- join_keys(
 )
 ```
 
-#### Shiny app
+#### Step 5/5 - Creating the Shiny App
 
-Finally, we include `merge_ui` and `merge_srv` to the UI and server component of the `shinyApp`, respectively,
+Finally, we include `merge_ui` and `merge_srv` in the UI and server components of the `shinyApp`, respectively,
 using the `data_extract`s defined in the first code block and the `datasets` object:
```
Comment (chlebowa) [inline vignettes/data-extract-merge.Rmd:165]:
```suggestion
Finally, we include appropriate `merge_ui` and `merge_srv` calls in the UI and server of our `shiny` app, respectively.
```

File: vignettes/data-extract.Rmd [hunk 17]
```diff
@@ -18,20 +18,19 @@ knitr::opts_chunk$set(
 ```
 
 There are times when an app developer wants to offer users more flexibility in analyzing data within their custom module. 
-In such cases, relinquishing control of the application to users requires developers to provide a degree of freedom. 
 With `teal`, app developers can open up their applications to users, allowing them to decide exactly which app data to 
 analyze within the module.
```
Comment (chlebowa) [inline vignettes/data-extract.Rmd:22]: Still some redundancy here.

File: vignettes/data-merge.Rmd [hunk 18]
```diff
@@ -17,16 +17,14 @@ knitr::opts_chunk$set(
 )
 ```
 
-Combining datasets is an essential step when working with modules that involve multiple datasets.
+Combining datasets is an essential step when working with multiple datasets.
```
Comment (chlebowa) [inline vignettes/data-merge.Rmd:20]:
```suggestion
Combining datasets is an essential step when working with relational datasets.
```
datasets -> data models?

Comment (chlebowa) [inline vignettes/data-merge.Rmd:20]: There is a `merge` function in `base`, I don't think we have to explain what we mean by merging. The next sentence could be skipped.

Alternatively, start with "joining" instead of "combining" as that is the standard SQL term.

File: vignettes/data-merge.Rmd [hunk 19]
```diff
@@ -17,16 +17,14 @@ knitr::opts_chunk$set(
...
 Within the context of `teal`, we use the term "merge" to refer to the process of combining datasets.
-To support this, two functions are provided: `merge_expression_module` and `merge_expression_srv`, each tailored for different use cases.
+To support this, two functions are provided:
 
-Use `merge_expression_module` when there is no need to process the `data_extract` list.
+1. `merge_expression_module` can be used when there is no need to process the `data_extract` list.
```
Comment (chlebowa) [inline vignettes/data-merge.Rmd:24]: `data_extract` is an arbitrary variable name, explain that it refers to a list of `des` objects.

File: vignettes/data-merge.Rmd [hunk 20]
```diff
@@ -105,7 +114,11 @@ merge_module <- function(id, datasets, data_extracts, join_keys) {
     output$data <- renderDataTable(ANL())
   })
 }
+```
+
+#### Step 4/5 - Preparing the Data
```
Comment (chlebowa) [inline vignettes/data-merge.Rmd:119]: As in the other vignette, creating the data may be a better place to start.

**Review by chlebowa — CHANGES_REQUESTED** [review]

File: vignettes/data-extract-merge.Rmd (hunk 15 above)
Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: It should be `merge_expression_srv`.
The `data_extract_srv`, `merge_expression_srv`, and `merge_expression_module` can be used in the module server along with `data_extract_ui` in the module UI.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: I'll remove it here. Looks like it was a copy paste from the other vignette. We already explained about `merge_expression_srv` above.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: Removed in both the places. They are completely redundant.

File: vignettes/data-merge.Rmd (hunk 20 above)
Comment (vedhav) [reply vignettes/data-merge.Rmd:119]: I'll follow this order.
Data -> DES creation -> Module UI + Server -> Shiny app

File: vignettes/data-merge.Rmd (hunk 19 above)
Comment (vedhav) [reply vignettes/data-merge.Rmd]: It should have been a list of `data_extract_spec`. I have changed it.

File: vignettes/data-extract.Rmd (hunk 17 above)
Comment (vedhav) [reply vignettes/data-extract.Rmd:22]: I think the first line can go.

File: vignettes/data-extract-merge.Rmd (hunk 13 above)
Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: I don't quite understand what you mean. Could you elaborate?

Comment (chlebowa) [reply vignettes/data-extract-merge.Rmd]: First you say 
```
1. Selecting the column `AGE` from `ADSL`
```
and then you show
```
adsl_extract <- data_extract_spec(
  dataname = "ADSL",
  select = select_spec(
    label = "Select variable:",
    choices = c("AGE", "BMRKR1"),
    selected = "AGE",
    multiple = TRUE,
    fixed = FALSE
  )
)
```

It is not obvious why I am only allowed to choose from `AGE` and `BMRKR1`.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: Okay, I see your point. I'll change it to `names(ADSL)` in all the places.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: Not in the extract vignette because we mention `select a variable from c("BMRKR1", "AGE")`

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: But wait a minute if we do this, The UI will also have the primary keys, which will not be removed even if they stay unselected. Which might be odd. Perhaps the best thing to do is to just mention that we also limit the choices to select from. But I think it's self-explanatory. What do you think?

Comment (chlebowa) [reply vignettes/data-extract-merge.Rmd]: We can mention that the app dev can limit the user's choice, yes. But take care that it does not distract from the core idea of the user selecting columns to keep.

As for keys, is it possible to not select them? They are automatically included when only `AGE` is selected, aren't they?

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: Made the `choices = c("AGE", "SEX", "BMRKR1")` for `ADSL` and `choices = c("AVAL", "AVALC", "ASEQ")` for `ADTTE` as the diagram simulates those are the available columns before the selection.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: Irrespective of the user selecting the key columns they will be selected by `teal.transform` so this feels like a bug for the app user.
I will add a comment about the ability to restrict the option choices, as it's an important feature to showcase.

Comment (vedhav) [reply vignettes/data-extract-merge.Rmd]: How does it look now?
 
Screenshot 2024-02-16 at 8 00 38 PM

**Review by chlebowa — APPROVED** [review]
Well done @vedhav :+1:

File: vignettes/data-extract.Rmd [hunk 21]
```diff
@@ -17,62 +17,25 @@ knitr::opts_chunk$set(
...
 
-Many `teal` modules leverage `data_extract_spec` objects and modules to handle user input.
-Examples can be found in `teal.modules.general` and `teal.modules.clinical`.
+A `teal` module can leverage the use of `data_extract_spec` objects to handle and process the user input.
+Examples can be found in the [modules from the `teal.modules.clinical` package](https://insightsengineering.github.io/teal.modules.clinical/latest-tag/reference/index.html).
 
 ### `data_extract_spec`
 
 The role of `data_extract_spec` is twofold: to create a UI component in a `shiny` application and to pass user input 
 from the UI to the module itself.
```
[the remark this answers is not in the document]
Comment (vedhav) [reply vignettes/data-extract.Rmd:29]: I agree that it was not clear here but after reading the whole vignette it becomes clear what this `module` means.
To keep it accurate and easier I'll rephrase it like this:
```
The role of `data_extract_spec` is twofold: to create a UI component in a `shiny` application and to pass user input 
from the UI to a custom server logic that can use this input to transform the data.
```

[the remark this answers is not in the document]
Comment (vedhav) [reply vignettes/data-extract.Rmd:29]: Thanks for the review @chlebowa that was a nice improvement suggestions along the way.
